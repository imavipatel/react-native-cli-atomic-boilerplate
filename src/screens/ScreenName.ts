export enum ScreenName {
  LoginScreen = 'LoginScreen',
  RegisterScreen = 'RegisterScreen',
  ProfileScreen = 'ProfileScreen',
}
