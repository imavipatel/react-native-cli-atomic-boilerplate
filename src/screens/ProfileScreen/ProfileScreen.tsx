import React from 'react';

import { StackNavigationProp } from '@react-navigation/stack';
import { ProfileTemplate } from 'components/templates';
import { RootStackParamList } from 'App';
import { ScreenName } from 'screens/ScreenName';
import { useNavigation } from '@react-navigation/native';

type ProfileScreenProps = {
  route: { params: { firstName: string; id: number; lastName: string } };
};

const ProfileScreen: React.FC<ProfileScreenProps> = ({ route }) => {
  const navigation = useNavigation<StackNavigationProp<RootStackParamList>>();

  const onClickLogoutLabel = () => {
    navigation.navigate(ScreenName.LoginScreen);
  };

  return (
    <ProfileTemplate
      onClickLogoutLabel={onClickLogoutLabel}
      title={`Hi ${route.params.firstName} ðŸ‘‹ you are logged in`}
    />
  );
};

export default ProfileScreen;
