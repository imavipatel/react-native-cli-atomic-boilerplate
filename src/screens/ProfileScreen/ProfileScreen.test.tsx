import React from 'react';
import { act, fireEvent, render, waitFor } from '@testing-library/react-native';
import ProfileScreen from './ProfileScreen';
import { ScreenName } from 'screens/ScreenName';

const mockNavigate = jest.fn();
const mockResetNavigate = jest.fn();
jest.mock('@react-navigation/native', () => ({
  useNavigation: (): { navigate: () => void; reset: () => void } => ({
    navigate: mockNavigate,
    reset: mockResetNavigate,
  }),
  useIsFocused: jest.fn(),
}));

describe('ProfileScreen', () => {
  const routerMock = {
    params: { id: 0, firstName: 'Lisa', lastName: 'Doe' },
  };
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const screen = render(<ProfileScreen route={routerMock} />);
    expect(screen).toMatchSnapshot();
  });

  it('should call the provided onSubmitButton function when pressed', async () => {
    const screen = <ProfileScreen route={routerMock} />;
    const wrapper = render(screen);

    fireEvent(wrapper.getByTestId('logout-label'), 'onPress');
    await act(async () =>
      waitFor(() =>
        expect(mockNavigate).toBeCalledWith(ScreenName.LoginScreen),
      ),
    );
  });
});
