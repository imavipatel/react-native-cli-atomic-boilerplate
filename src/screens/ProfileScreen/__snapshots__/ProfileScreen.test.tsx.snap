// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileScreen should render correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <View
    style={
      Object {
        "height": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 2,
          "flexDirection": "column",
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "minHeight": 45,
            "minWidth": 200,
          }
        }
        testID="header"
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 30,
              "fontWeight": "600",
            }
          }
          testID="header-text"
        >
          Hi Lisa ðŸ‘‹ you are logged in
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 2,
          "justifyContent": "center",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        testID="logout-label"
      >
        <Text
          style={
            Object {
              "color": "black",
              "fontSize": 16,
              "fontWeight": "600",
            }
          }
          testID="label"
        >
          Logout
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 4,
        }
      }
    />
  </View>
</RCTSafeAreaView>
`;
